lib/signer.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+
+/**
+ * @typedef {signer}
+ * @function sign
+ */
+
+
+/**
+ * Creates signer.
+ *
+ * @param {AuthorizationBuilder} AuthorizationBuilder
+ * @return {signer}
+ */
+module.exports = (AuthorizationBuilder) => {
+    /**
+     * @param {string} code
+     * @param {string} secret
+     * @param {requestOptions} requestOptions
+     * @param {string|stream~Readable} body
+     *
+     * @return {Promise}
+     */
+    function sign(code, secret, requestOptions, body) {
```

Comment:
Given how the rest of the code works, wouldn't you want to do something like

    // if string, convert string to a stream
    signerStream = new signer("stuff here")
    stream.pipe(signerStream).pipe(request)

That should work fine.  The current approach seems more confusing.